geode-core/src/test/java/org/apache/geode/internal/cache/tx/PartitionedTXRegionStubTest.java

```diff
@@ -127,6 +129,36 @@ public void destroyExistingEntryThrowsTransactionDataNodeHasDepartedExceptionIfI
     verify(stub, never()).trackBucketForTx(keyInfo);
   }
 
+  @Test
+  public void getEntryForIteratorReturnsEntryGotFromLocal() throws Exception {
+    PartitionedTXRegionStub stub = spy(new PartitionedTXRegionStub(txStateStub, partitionedRegion));
+    EntrySnapshot entry = mock(EntrySnapshot.class);
+
+    when(event.getRegion()).thenReturn(partitionedRegion);
+    when(partitionedRegion.getBucketPrimary(1))
+        .thenReturn((InternalDistributedMember) remoteTransactionHost);
+    when(partitionedRegion.getEntryRemotely((InternalDistributedMember) remoteTransactionHost, 1,
+        key, false, true)).thenReturn(entry);
+
+    assertThat(stub.getEntryForIterator(keyInfo, true)).isEqualTo(entry);
+    verify(stub).trackBucketForTx(keyInfo);
+  }
+
+  @Test
+  public void getEntryForIteratorReturnsEntryGotFromRemote() throws Exception {
```

Comment:
Similarly getEntryForIteratorReturnsEntryGotFromAnyDataHosts is more appropriate here. 
Also the method does not throw Exception, please remove it to avoid warnings.